code/stata-whitespace-columns.do

```diff
@@ -1,15 +1,17 @@
-* This is BAD
+ACCEPTABLE:
+
     * Create dummy for being employed
     generate employed = 1
-    replace employed = 0 if (_merge == 2)
-    label variable employed "Person exists in employment data"
-    label define yesno 1 "Yes" 0 "No"
-    label value employed yesno
+        replace employed = 0 if (_merge == 2)
+        lab var employed "Person exists in employment data"
+        lab def yesno 1 "Yes" 0 "No"
+        lab val employed yesno
+
+BETTER:
 
-* This is GOOD
     * Create dummy for being employed
-    generate       employed = 1
-    replace        employed = 0 if (_merge == 2)
-    label variable employed "Person exists in employment data"
-    label define            yesno 1 "Yes" 0 "No"
-    label value    employed yesno
+    generate    employed = 1
+        replace employed = 0 if (_merge == 2)
+        lab var employed "Person exists in employment data"
+        lab def          yesno 1 "Yes" 0 "No"
+        lab val employed yesno
```

Comment:
I remain unconvinced of this extra tab. I think the comment is already indicating where the code chunk starts

Reply:
You mean for lines beyond the first after the comment, ie are you saying all these lines should be at the same indentation?  In that case automated indent guides don't work, but you're right that the block is then visible as such. Could go either way on this, as the guiding principle is the visual clarity/separation of the bigger task

Reply:
Agreed all three in person for b4a69a9469543303ae213d36a116d0a5ced571e0